substrate/frame/balances/src/lib.rs

```diff
@@ -1248,5 +1278,43 @@ pub mod pallet {
 			});
 			Ok(actual)
 		}
+
+		/// Generate dev account from derivation string.
+		#[cfg(feature = "runtime-benchmarks")]
+		pub fn derive_dev_account(num_accounts: u32, balance: T::Balance, derivation: &String) {
+			// Ensure that the number of accounts is not zero
+			assert!(num_accounts > 0, "num_accounts must be greater than zero");
+
+			assert!(
+				balance >= <T as Config<I>>::ExistentialDeposit::get(),
+				"the balance of any account should always be at least the existential deposit.",
+			);
+
+			assert!(
+				derivation.contains("{}"),
+				"Invalid derivation string"
+			);
+
+			for index in 0..num_accounts {
+				// Replace "{}" in the derivation string with the index.
+				let derivation_string = derivation.replace("{}", &index.to_string());
+
+				// Attempt to create the key pair from the derivation string with error handling.
+				let pair: SrPair = Pair::from_string(&derivation_string, None)
+					.expect(&format!("Failed to parse derivation string: {}", derivation_string));
+
+				// Convert the public key to AccountId.
+				let who = T::AccountId::decode(&mut &pair.public().encode()[..])
+					.expect(&format!("Failed to decode public key from pair: {:?}", pair.public()));
```

Comment:
```suggestion
					.unwrap_or_else(|| format!("Failed to parse derivation string: {derivation_string}"));

				// Convert the public key to AccountId.
				let who = T::AccountId::decode(&mut &pair.public().encode()[..])
					.unwrap_or_else(|| format!("Failed to decode public key from pair: {:?}", pair.public()));
```

Reply:
This would expect a default value for the respective operations, `.expect()` can't be used at genesis?